packages/docusaurus-theme-classic/src/theme/Showcase/ShowcaseCard/index.tsx

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import clsx from 'clsx';
+import Link from '@docusaurus/Link';
+import Translate from '@docusaurus/Translate';
+import {sortBy} from '@docusaurus/plugin-content-showcase/client';
+import {
+  useShowcaseTags,
+  useShowcaseApiScreenshot,
+} from '@docusaurus/theme-common/internal';
+import Heading from '@theme/Heading';
+import FavoriteIcon from '@theme/Showcase/FavoriteIcon';
+import type {ShowcaseItem, TagType} from '@docusaurus/plugin-content-showcase';
+import styles from './styles.module.css';
+
+function TagItem({
+  label,
+  description,
+  color,
+}: {
+  label: string;
+  description: {
+    message: string;
+    id: string;
+  };
+  color: string;
+}) {
+  return (
+    <li className={styles.tag} title={description.message}>
+      <span className={styles.textLabel}>{label.toLowerCase()}</span>
+      <span className={styles.colorLabel} style={{backgroundColor: color}} />
+    </li>
+  );
+}
+
+// TODO move tag reorder logic into hook
+function ShowcaseCardTag({tags}: {tags: TagType[]}) {
+  const Tags = useShowcaseTags();
+  const TagList = Object.keys(Tags) as TagType[];
+
+  const tagObjects = tags.map((tag) => ({tag, ...Tags[tag]}));
+
+  // Keep same order for all tags
+  const tagObjectsSorted = sortBy(tagObjects, (tagObject) =>
+    TagList.indexOf(tagObject.tag),
+  );
```

Comment:
The plugin should probably short the site tags on the nodejs side, this way the theme receives already-sorted tags and we don't need to run any extra runtime logic